mimic/test/dummy.py

```diff
@@ -195,7 +203,17 @@ def make_example_external_api(case, name=u"example",
     if endpoint_templates is None:
         endpoint_templates = [ExampleEndpointTemplate()]
 
-    service_type = endpoint_templates[0].type_key
+    if service_type is None:
+        # default parameter value, take the template type from the first
+        # endpoint template in the list
+        service_type = endpoint_templates[0].type_key
+
+    elif len(endpoint_templates) == 1:
+        # service_type was specified and most likely endpoint_templates
```

Comment:
"most likely" seems a bit weird of an assumption to make, especially if you're going to be mutating the value passed in `endpoint_templates` in this specific case; if that's the intent, why not remember whether `endpoint_templates is None` earlier when you tested for it.  Was this written backwards?